script.js

```diff
@@ -1,4 +1,586 @@
+//   if (
+//     (userInput == `scissors` && programSelection == `stone`) ||
+//     (userInput == `stone` && programSelection == `paper`) ||
+//     (userInput == `paper` && programSelection == `scissors`)
+//   ) {
+//     message = `You won! <br> <br> This is the reversed game. <br> <br> You chose ${userInput} and I chose ${programSelection}. <br> Good job!`;
+//   }
+
+//   return message;
+// };
+
+// // Specify what happens for the 3 game modes
+// var main = function (input) {
+//   var myOutputValue = ``;
+
+//   if (currentGameMode == `waiting for player name`) {
+//     playerName = input;
+//     currentGameMode = `standard game`;
+//     myOutputValue = `Welcome, ${playerName}! <br> Please type 'scissors', 'paper' or 'stone' to start playing the game with me 😊 <br> <br> You can also type 'reversed' to try our reversed mode and 'standard' to revert back to the original game.`;
+//   } else if (currentGameMode != `waiting for player name`) {
+//     currentGameMode = changeGameMode(input);
+//     programChoice = generateChoice();
+//     if (currentGameMode == `standard game`) {
+//       myOutputValue = standardGame(input, programChoice);
+//     }
+//     if (currentGameMode == `reversed game`) {
+//       myOutputValue = reversedGame(input, programChoice);
+//     }
+//   }
+//   return myOutputValue;
+// };
+
+//======SPS Game with Korean Game Mode: Project Part 2 (More Comfortable)=======
+// Specify global variables
+var currentGameMode = `waiting for player name`;
+var lastWonPlayer = `unassigned`;
+
+// Create a function to change game mode based on input
+var changeGameMode = function (userInput) {
+  var gameMode = currentGameMode;
+  if (userInput == `standard`) {
+    gameMode = `standard game`;
+  }
+  if (userInput == `korean`) {
+    gameMode = `korean game`;
+  }
+  console.log(`game mode is ${gameMode}`);
+  return gameMode;
+};
+
+// Create a function to generate random scissors/paper/stone choices
+var generateChoice = function () {
+  var randomDecimal = Math.random() * 3;
+  var randomInteger = Math.floor(randomDecimal);
+  if (randomInteger == 0) {
+    assignedChoice = `scissors`;
+  }
+  if (randomInteger == 1) {
+    assignedChoice = `paper`;
+  }
+  if (randomInteger == 2) {
+    assignedChoice = `stone`;
+  }
+  return assignedChoice;
+};
+
+// Create a function to derive the result of a standard SPS game
+var result = function (userInput, programSelection) {
+  var gameResult = ``;
+  if (userInput == programSelection) {
+    gameResult = `draw`;
+  }
+  if (
+    (userInput == `scissors` && programSelection == `paper`) ||
+    (userInput == `paper` && programSelection == `stone`) ||
+    (userInput == `stone` && programSelection == `scissors`)
+  ) {
+    gameResult = `win`;
+  }
+  if (
+    (userInput == `scissors` && programSelection == `stone`) ||
+    (userInput == `stone` && programSelection == `paper`) ||
+    (userInput == `paper` && programSelection == `scissors`)
+  ) {
+    gameResult = `lose`;
```

Comment:
Great that you have a clear understanding on how the if else loops runs, and this being the default result being assigned to `gameResult`